### PR TITLE
Quell warnings

### PR DESCRIPTION
**Pull request by notspiff**
Is that cast needed for some reason I do not see?

### CODE REVIEW

**Comment by Montellese** [comment]
Can't think of any reason. First I thought it might be that the expressing (including the `- 1` can become negative but `units` has a fixed positive size so that can't be it either.

File: xbmc/utils/StringUtils.cpp [hunk 1]
```diff
@@ -1295,7 +1295,7 @@ std::string StringUtils::FormatFileSize(uint64_t bytes)
 
   unsigned int i = 0;
   double value = static_cast<double>(bytes);
-  while (i < static_cast<int>(units.size()) - 1 && value >= 999.5)
+  while (i < units.size() - 1 && value >= 999.5)
```
Comment (garbear) [inline xbmc/utils/StringUtils.cpp]: Should be `while (i + 1 < units.size() && ...`

File: xbmc/addons/settings/AddonSettings.cpp [hunk 2]
```diff
@@ -1350,15 +1350,15 @@ bool CAddonSettings::ParseOldCondition(std::shared_ptr<const CSetting> setting,
         case SettingOptionsType::Static:
         {
           const auto& options = referencedSettingString->GetOptions();
-          if (options.size() > valueIndex)
+          if (options.size() > static_cast<size_t>(valueIndex))
```
Comment (garbear) [inline xbmc/addons/settings/AddonSettings.cpp]: valueIndex should be made unsigned

Comment (Montellese) [inline xbmc/addons/settings/AddonSettings.cpp]: Hm I already have this fixed somewhere...

File: xbmc/utils/StringUtils.cpp (hunk 1 above)
Comment (Paxxi) [inline xbmc/utils/StringUtils.cpp]: i should be std::size_t and not unsigned int otherwise it'll generate warnings as size_t can be 32/64 bit

Comment (notspiff) [inline xbmc/utils/StringUtils.cpp]: done.

File: xbmc/addons/settings/AddonSettings.cpp (hunk 2 above)
Comment (notspiff) [inline xbmc/addons/settings/AddonSettings.cpp]: done.

**Comment by notspiff** [comment]
whadya know. found another one hiding within the ffmpeg deprecation warnings.

**Comment by notspiff** [comment]
rebased. last time. i will just close on next conflict.

**Pull request merged**